src/Type/Constant/ConstantArrayType.php

```diff
@@ -216,6 +216,8 @@ public function getAllArrays(): array
 		} else {
 			$optionalKeysCombinations = [
 				[],
+				array_slice($this->optionalKeys, 0, 1, true),
+				array_slice($this->optionalKeys, -1, 1, true),
 				$this->optionalKeys,
 			];
```

Comment:
When an array only have optionalKey AND is non-empty, the result of getAllArray intersected with NonEmptyArray will give an array with all the keys and no key will be reported with
```
$innerType->hasOffsetValueType($innerDimType)->no()
```

So I have to
- add a smallest array possible to report every optional key (except the first one)
- add another smallest array possible to report the first optional key